File: net/JetBrains.Serialization/src/Dataholders/IdCtTSTInfo.cs
using Newtonsoft.Json;
using Org.BouncyCastle.Asn1;
using Org.BouncyCastle.Asn1.Cms;

namespace JetBrains.Serialization;

[JsonObject(MemberSerialization.OptIn)]
public class IdCtTSTInfo : EncapContentInfo
{
  [JsonProperty("ContentType")] protected override TextualInfo ContentType { get; }

  [JsonProperty("Content")] private TextualInfo _content;

  [JsonConstructor]
  public IdCtTSTInfo(TextualInfo contentType, TextualInfo content)
  {
    ContentType = contentType;
    _content = content;
  }

  public IdCtTSTInfo(ContentInfo contentInfo)
    : this(TextualInfo.GetInstance(contentInfo.ContentType), TextualInfo.GetInstance(contentInfo.Content))
  {
  }

  protected override Asn1Encodable GetContentPrimitive() => _content.ToPrimitive();
}

File: jvm/src/main/kotlin/com/jetbrains/signatureverifier/serialization/dataholders/IdCtTSTInfo.kt
package com.jetbrains.signatureverifier.serialization.dataholders

import kotlinx.serialization.Serializable
import org.bouncycastle.asn1.cms.ContentInfo

@Serializable
data class IdCtTSTInfo(
  override val contentType: TextualInfo,
  val content: TextualInfo
) : EncapContentInfo() {

  constructor(contentInfo: ContentInfo) : this(
    TextualInfo.getInstance(contentInfo.contentType),
    TextualInfo.getInstance(contentInfo.content)
  )

  override fun getContentPrimitive() = content.toPrimitive()
}